src/test/java/com/devonfw/sample/archunit/ThirdPartyRulesE3JpaCheckTest.java

```diff
@@ -0,0 +1,50 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class ThirdPartyRulesE3JpaCheckTest {
+
+    private static boolean isUsingJavaxPersistenceDataAccessOrEmbeddablesInCommon(JavaClass item, String targetPackageFullName) {
+        if(targetPackageFullName.startsWith("javax.persistence")) {
+            if(item.getFullName().contains("dataaccess")) {
+                return true;
+            }
+            if(item.getFullName().contains("common") && item.getSimpleName().contains("Embeddable")) {
```

Comment:
it would be better to reuse our regular expression and have some reusable code to check layer and other aspects of the package. Especially this rule would also apply in case someone has an entity `com.customer.app.commoncomponent.dataaccess.MyCommonEntity` as the qualified name contains `common`.

Reply:
Concerning #19, we should discuss with the team how we want to support the reusing of given regular expressions.

For the moment, I have added two `public static final` patterns for the common and dataaccess layers into `PackageRuleTest.java`. I utilize these patterns to check whether the source class lies inside the common or dataaccess layer instead of that faulty implementation.

Reply:
same here. We can merge now, but has to be reworked anyhow. `StructuredPackage` now available.